code/__HELPERS/~monkestation-helpers/roundend.dm

```diff
@@ -36,3 +36,26 @@
 			total_payout += listed_challenge.challenge_payout
 		if(total_payout)
 			client?.prefs?.adjust_metacoins(client?.ckey, total_payout, "Challenge rewards.")
+
+/datum/controller/subsystem/ticker/proc/refund_cassette()
+	if(!GLOB.cassette_reviews || !islist(GLOB.cassette_reviews) || !GLOB.cassette_reviews.len)
+		return
+
+	for(var/id in GLOB.cassette_reviews)
+		var/datum/cassette_review/review = GLOB.cassette_reviews[id]
+		if(!review || review.action_taken) // Skip if review doesn't exist or already handled (denied / approved)
+			continue
+
+		var/ownerckey = review.submitted_ckey // ckey of who made the cassette.
+		if(!ownerckey)
+			continue
+
+		var/client/client = GLOB.directory[ownerckey] // Use directory for direct lookup (Client might be a differnet mob than when review was made.)
+		if(client && !QDELETED(client?.prefs))
+			var/adjusted = client?.prefs?.adjust_metacoins(
+				client?.ckey, 5000,
+				reason = "No action taken on cassette:\[[review.submitted_tape.name]\] before round end.",
+				announces = TRUE, donator_multipler = FALSE
+			)
+			if(adjusted)
+				review.Destroy()
```

Comment:
```suggestion
				qdel(review)
```
never directly call destroy - qdel calls that.

Reply:
Ah this may have solved that runtime issue.

Reply:
Ah I see still have an issue with the cassette_review/Destroy() doing a bad qdel But this bit I did resolve. runtime error: bad del
 - proc name: qdel (/proc/qdel)
 -   source file: code/controllers/subsystem/garbage.dm,364
 -   usr: (src)
 -   src: null
 -   call stack:
 - qdel(null, 0)
 - /datum/cassette_review (/datum/cassette_review): Destroy(0)
 - qdel(/datum/cassette_review (/datum/cassette_review), 0)
 - Ticker (/datum/controller/subsystem/ticker): refund cassette()
 - Ticker (/datum/controller/subsystem/ticker): declare completion(1)